pandas/core/algorithms.py

```diff
@@ -1685,6 +1688,10 @@ def map_array(
     na_action : {None, 'ignore'}, default None
         If 'ignore', propagate NA values, without passing them to the
         mapping correspondence.
+    convert : bool, default True
+        Try to find better dtype for elementwise function results. If
+        False, leave as dtype=object. Note that the dtype is always
+        preserved for some extension array dtypes, such as Categorical.
```

Comment:
`convert=True` is already the default in `map_infer`and `map_infer_mask`, so this doesn't change anything. 

Wrt. EA.map, this PR changes nothing compared to the status quo as mentioned, and actually keeps the same behavior as in pandas v1.5. Possibly we should make EA.map more intelligent, but that is a different question and could be maybe be done by changing `ExtensionArray.map`.

I did propose some changed behavior for datetimelikes.map in #51936, but that is an independent PR.

Reply:
Sorry, I get what you asked now, you're asking if the doc string may be wrong? I actually think so, because we're converting to object dtype here yes, but let me take a look.

Reply:
Yeah I'm mostly interested if this is going to have implications in the new EA.map with convert=True since EA.map by default calls `map_array`

Reply:
If you look into the `_map_values` method, you can see it won't affect EA.map. I can update the doc string.